### DISCUSSION TITLE
Component with data in a EcsWith pair ?

### DISCUSSION THREAD

**Discussion by sigzegv** [SanderMertens/flecs#1146] [Q&A] [answered]
I am trying to create an EcsWith pair with a component with default data like this

```
    typedef struct {
        int val;
    } Responsability;

    [...]

    ECS_COMPONENT(w, Responsability);

    [...]

    ecs_entity_t Power = ecs_new_id(w);
    ecs_set_pair_second(w, Power, EcsWith, Responsability, {5});
```

Seems not possible as it fails at runtime with `fatal: flecs.c: 4419: assert: r->table->column_count != 0 INTERNAL_ERROR`

It is possible to do this in some way ?

**Comment by copygirl** [comment, marked as answer]
`EcsWith` is a registered as a tag, meaning a pair with it being the relationship type can't have a value associated with it.

As a feature request, maybe there could be an alternative `EcsWithDefault` that does accept a value.

Alternatively, I believe you could use a component hook to initialize `Responsability` with the default value you'd like.

I can't speak on whether the error is intentional in this case. Sounds like the job for an assert?
